tests/CMakeLists.txt

```diff
@@ -1,4 +1,57 @@
-# Add subdirectories of all the tests
+# ----------------------------------------------
+# Python requirements
+# ----------------------------------------------
+find_package(PythonInterp)
+if(NOT PYTHONINTERP_FOUND)
+  message(
+    FATAL_ERROR
+      "You need python installed to generate unittests. If you do not want to compile the unittests pass -DBUILD_TESTING=OFF to cmake."
+  )
+endif()
+
+include(FindPythonModule)
+find_python_module(numpy)
+if(NOT NUMPY_FOUND)
+  message(
+    FATAL_ERROR
+      "You need numpy python module installed to generate unittests. If you do not want to compile the unittests pass -DBUILD_TESTINGOFF to cmake."
+  )
+endif()
+
+find_python_module(scipy)
+# Check scipy version for sparse.random functionalities
+if((NOT SCIPY_FOUND) OR (SCIPY_VERSION VERSION_LESS 0.17.0))
+  message(
+    FATAL_ERROR
+      "You need scipy python module installed to generate unittests. If you do not want to compile the unittests pass -DBUILD_TESTING=OFF to cmake."
+  )
+endif()
+
+find_python_module(__future__)
```

Comment:
I think we don't need the `future` Python module as we have dropped Python 2.7 support.

Reply:
Related: we should remove the first 3 lines in `tests/utils/codegen_utils.py`.

Reply:
Ok, I fixed this in https://github.com/osqp/osqp/pull/352/commits/54ef9e9d5722cbf7f3ff57fab49439921ceed965